tracardi/service/storage/drivers/elastic/event.py

```diff
@@ -222,15 +222,23 @@ def convert_data(raw_result):
     return list(convert_data(raw_result))
 
 
-async def update_field(event_type: str, field: str, value: Any):
+async def update_tags(event_type: str, tags: List[str]):
     query = {
         "script": {
-            "source": f"ctx._source.{field} = {value};",
+            "source": f"ctx._source.tags.content = {tags}; ctx._source.tags.count = {len(tags)}",
             "lang": "painless"
         },
         "query": {
-            "match": {
-                "type": event_type
+            "bool": {
+                "must": {"match": {"type": event_type}},
+                "must_not": {
+                    "bool": {
+                        "must": [
+                            *[{"term": {"tags.content": tag}} for tag in tags],
+                            {"term": {"tags.count": len(tags)}}
```

Comment:
Why is count used?

Reply:
What is the use case for that?

Reply:
@atompie 
We do not want to update the events that already have wanted tags. We want these tags to match exactly though and elasticsearch does not provide way to efficiently compare arrays. I've read some documentations and other people's issues and adding count is considered the fastest and cheapest way to make exact matches on arrays. It allows us to check for an equality, not only check if one array contains another one.
For example:
Our tags are ["tag1", "tag2"]
Without count:
Event having tags ["tag1", "tag2", "tag3"] is considered up-to-date, but in fact the tags are different.
With count:
["tag1", "tag2", "tag3"] requires updating, since array length (our counter) is not equal to 2.

Hope this helps,
Let mi know if you have better ways to do that.